repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2980,57 +2976,91 @@ private void updateInConsistentOwnedMapVertices(AttributeMutationContext ctx, At
-                classificationText = fullTextMapperV2.getClassificationTextForEntity(new AtlasEntity(entityHeader));
-                if(classification.isPropagate() && !classification.getEntityGuid().equals(entityHeader.getGuid())) {
-                    //Add to propagatedClassificationNamesList
-                    propagatedClassificationNamesList.add(typeName);
-                    //Create delimiter separated string for propagatedClassificationNames
-                    if (StringUtils.isEmpty(propagatedClassificationName)) {
-                        propagatedClassificationName = CLASSIFICATION_NAME_DELIMITER + typeName + CLASSIFICATION_NAME_DELIMITER;
-                    } else {
-                        propagatedClassificationName = propagatedClassificationName + typeName + CLASSIFICATION_NAME_DELIMITER;
-                    }
-                } else {
-                    //Add to classificationNamesList
-                    classificationNamesList.add(typeName);
-                    //Create delimiter separated string for classificationNames
-                    if (StringUtils.isEmpty(classificationNames)) {
-                        classificationNames = CLASSIFICATION_NAME_DELIMITER + typeName + CLASSIFICATION_NAME_DELIMITER;
-                    } else {
-                        classificationNames = classificationNames + typeName + CLASSIFICATION_NAME_DELIMITER;
+    public void cleanUpClassificationPropagation(String classificationName) throws AtlasBaseException {
+        int batchSize = 100;
+        int counter = 0;
+        while (true) {
+            try {
+
+                List<AtlasVertex> vertices = GraphHelper.getAllAssetsWithClassificationAttached(graph, classificationName, batchSize);
+                if (CollectionUtils.isEmpty(vertices)) {
+                    LOG.info("No entities found for classification {}", classificationName);
+                    break;
+                }
+                for(AtlasVertex vertex : vertices) {
+                    String guid = GraphHelper.getGuid(vertex);
+                    GraphTransactionInterceptor.lockObjectAndReleasePostCommit(guid);
+                    List<AtlasClassification> deletedClassifications = new ArrayList<>();
+                    List<AtlasEdge> classificationEdges = GraphHelper.getClassificationEdges(vertex, null, classificationName);
+
+                    for (AtlasEdge edge : classificationEdges) {
+                        AtlasClassification classification = entityRetriever.toAtlasClassification(edge.getInVertex());
+                        deletedClassifications.add(classification);
+                        deleteDelegate.getHandler().deleteEdgeReference(edge, TypeCategory.CLASSIFICATION, false, true, null, vertex);
                     }
+
+                    AtlasEntity entity = repairClassificationMappings(vertex);
+
+                    entityChangeNotifier.onClassificationDeletedFromEntity(entity, deletedClassifications);
```

Comment:
is this the exact same notification that's sent when the classification is normally remove from an entity (both direct and propagation). Verify entity audits as well.

Reply:
Verified : For notification it just publish classification deletion event. But in case of audit it calculate during publish time if this propagated or not, so it correctly send audits.